publisher/planned_post.py

```
@@ -0,0 +1,178 @@
+                sched = self.schedule
+                sched.next_run = self.effective_datetime
+                sched.save()
+        else:  # новый объект
+            try:
+                with transaction.atomic():
+                    super().save(*args, **kwargs)
+                    self.status = 'scheduled'
+                    self.schedule = schedule_task(self)
+                    super().save(update_fields=['status', 'schedule'])
+            except Exception as e:
+                self.message_user(request, f"Ошибка при планировании поста: {str(e)}", level=messages.ERROR)
```

Ruff (0.12.2)

Undefined name `request`

(F821)